File: SPOS lab/SPOS lab/scheduling/priority.java
import java.util.*;
public class priority {
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        System.out.println("enter number of process: ");
        int n = sc.nextInt();
        int pid[] = new int[n];
        int pri[] = new int[n];
        int ar[] = new int[n];
        int bt[] = new int[n];
        int ct[] = new int[n];
        int ta[] = new int[n];
        int wt[] = new int[n];
        float avgwt = 0, avgta = 0;
        for (int i = 0; i < n; i++) {
            System.out.println("enter process " + (i + 1) + " arrival time");
            ar[i] = sc.nextInt();
            System.out.println("enter process " + (i + 1) + " burst time");
            bt[i] = sc.nextInt();
            System.out.println("enter process " + (i + 1) + "priority");
            pri[i]= sc.nextInt();
            pid[i] = i + 1;
        }
        sortarr(ar,bt,pri,pid,n);

        for (int i = 0; i < n; i++) {
            if (i == 0)
                ct[i] = ar[i] + bt[i];
            else if (ar[i] > ct[i - 1])
                ct[i] = ar[i] + bt[i];
            else
                ct[i] = ct[i - 1] + bt[i];
            ta[i]=ct[i]-ar[i];
            wt[i]=ta[i]-bt[i];
            avgta+=ta[i];
            avgwt+=wt[i];
        }
        System.out.println("\npid priority arrival brust complete turn waiting");
        for(int i = 0; i < n; i++)
        System.out.println(pid[i]+ " \t "+pri[i]+" \t"+ar[i]+" \t"+bt[i]+" \t"+ct[i]+" \t"+ta[i]+" \t"+wt[i]);
        sc.close();
        System.out.println("\naverage waiting time: "+(avgwt/n));
        System.out.println("\naverage turn around time: "+(avgta/n));
    }
    static void sortarr(int at[],int bt[],int pri[],int pid[],int n){
        int temp;
        for(int i=0;i<n;i++){
            for(int j=0;j<n-(i+1);i++){
                if(at[j]>at[j+1]){
                    temp=at[j];
                    at[j]=at[j+1];
                    at[j+1]=temp;

                    temp=bt[j];
                    bt[j]=bt[j+1];
                    bt[j+1]=temp;

                    temp=pri[j];
                    pri[j]=pri[j+1];
                    pri[j+1]=temp;

                    temp=pid[j];
                    pid[j]=pid[j+1];
                    pid[j+1]=temp;

                }
                if(at[j]==at[j+1]){
                    if(pri[j]>pri[j+1])
                    {
                        temp = at[j];
                        at[j] = at[j + 1];
                        at[j + 1] = temp;

                        //swapping burst time
                        temp = bt[j];
                        bt[j] = bt[j + 1];
                        bt[j + 1] = temp;

                        //swapping priority
                        temp = pri[j];
                        pri[j] = pri[j + 1];
                        pri[j + 1] = temp;

                        temp=pid[j];
                        pid[j]=pid[j+1];
                        pid[j+1]=temp;

                    }
                }
            }
        }
    }
}

File: SPOS lab/SPOS lab/schedulingcpp/priority.cpp
#include<bits/stdc++.h>
using namespace std;
class Custom{
    public:
    int pid;
    int at;
    int bt;
    int ct;
    int tat;
    int wt;
    int rt;
    int pl;
    
};

int main(){
    cout<<"Non premtive priority scheduling\n";
    cout<<"Enter the number of process\n";
    int n;
    cin>>n;
    Custom a[n];
    for(int i=0;i<n;i++){
        cout<<"Enter the at:- \n";
        cin>>a[i].at;
        cout<<"Enter the bt:- \n";
        cin>>a[i].bt;
        a[i].pid = i+1;
        cout<<"Enter the priority\n";
        cin>>a[i].pl;
    }
    int visited[n];
    for(int i=0;i<n;i++){
        visited[i] = 0;
    }
    int starting_time = 0;
    int count =0;
    while(count <n){
            int idx =-1;
            int pr = INT_MAX;
        for(int i=0;i<n;i++){
            if((a[i].at<=starting_time) && (!visited[i])){
                if(pr>a[i].pl){
                pr = a[i].pl;
                idx = i;
                }
            }
        }
        if(idx!=-1){
            visited[idx] = 1;
            a[idx].ct = starting_time + a[idx].bt;
            a[idx].tat = a[idx].ct - a[idx].at;
            a[idx].wt = a[idx].tat - a[idx].bt;
            a[idx].rt = starting_time - a[idx].at;
            count ++;
            starting_time += a[idx].bt;
        }
        // cout<<count<<" \n";
    }cout<<"ID\t At\t Bt\t pl\t Ct\t tat\t wt\t rt\n";
    int avgwt=0,avgtat=0;
    for(int i=0;i<n;i++){
        avgwt+=a[i].wt;
        avgtat+=a[i].tat;
        cout<<a[i].pid<<"\t"<<a[i].at<<"\t"<<a[i].bt<<"\t"<<a[i].pl<<"\t"<<a[i].ct<<"\t"<<a[i].tat<<"\t"<<a[i].wt<<"\t"<<a[i].rt<<"\n";
    }
    cout<<"Average waiting time "<<avgwt*1.0/n<<endl;
	cout<<"Average turn around time "<<avgtat*1.0/n<<endl;;
    return 0;
}